### ISSUE TITLE
v1.4 on GKE - "object doesn't exist"

### ISSUE THREAD

**Issue by dwhitena** [pachyderm/pachyderm#1530]
Reported from a user on slack using 1.4 on GKE.  He started from a fresh installation and:

```
$ pachctl delete-all
...
$ pachctl create-repo gene_expression
$ pachctl put-file gene_expression master /ge.txt -c -f /mnt/c/temp/exprs.txt
$ pachctl get-file gene_expression master /ge.txt
storage: object doesn't exist
$ pachctl list-file gene_expression master /
storage: object doesn't exist
```

**Comment by dwhitena** [comment]
Yuval Kalugny [10:08 AM] 
@dwhitena
1. This was a fresh installation, new cluster and all. I actually deleted everything and tried again from scratch.
2. The `list-file` is in the examples above (same error as `get-file`). `list-repo` worked as expected (I don't have that deployment anymore so no screenshots).
One thing that may be relevant - when installing 1.4 I had the `etcd` pod fail due to a timeout, something to do with disk permissions (a shame I didn't record it). I deleted the pod and when it was re-created it worked so I thought nothing of it.

**Comment by dwhitena** [comment]
Another user experiencing this issue:

> Hello again, I keep receiving the error message `storage: object doesn't exist` when a try to list or get the files from a directory. Anyone with the same problem?
> (Using pachyderm 1.4, deployed at Google Cloud)

**Comment by dwhitena** [comment]
Ok, I have tried to reproduce this.  I deployed a GKE 1.4 cluster as specified below.  I did have one problem, which may or may not be related.  When I deploy pachyderm the first time, etcd seemed to not come up.  However, I did `pachctl undeploy`, then `pachctl deploy ...` again, and it came up fine.  

```sh
➔ CLUSTER_NAME=pach-test-bug
➔ gcloud container clusters create ${CLUSTER_NAME} --scopes storage-rw
Creating cluster pach-test-bug...done.                                                                                                                                                  
Created [https://container.googleapis.com/v1/projects/dan-dev-155821/zones/us-west1-a/clusters/pach-test-bug].
kubeconfig entry generated for pach-test-bug.
NAME           ZONE        MASTER_VERSION  MASTER_IP       MACHINE_TYPE   NODE_VERSION  NUM_NODES  STATUS
pach-test-bug  us-west1-a  1.5.6           35.185.241.196  n1-standard-1  1.5.6         3          RUNNING
➔ gcloud container clusters get-credentials ${CLUSTER_NAME}
Fetching cluster endpoint and auth data.
kubeconfig entry generated for pach-test-bug.
➔ STORAGE_SIZE="10"
➔ STORAGE_NAME=pach-bug-test
➔ gcloud compute disks create --size=${STORAGE_SIZE}GB ${STORAGE_NAME}
WARNING: You have selected a disk size of under [200GB]. This may result in poor I/O performance. For more information, see: https://developers.google.com/compute/docs/disks#pdperformance.
Created [https://www.googleapis.com/compute/v1/projects/dan-dev-155821/zones/us-west1-a/disks/pach-bug-test].
NAME           ZONE        SIZE_GB  TYPE         STATUS
pach-bug-test  us-west1-a  10       pd-standard  READY

New disks are unformatted. You must format and mount a disk before it
can be used. You can find instructions on how to do this at:

https://cloud.google.com/compute/docs/disks/add-persistent-disk#formatting

➔ BUCKET_NAME=pach-bug-test
➔ gsutil mb gs://${BUCKET_NAME}
Creating gs://pach-bug-test/...
➔ pachctl deploy google ${BUCKET_NAME} ${STORAGE_SIZE} --static-etcd-volume=${STORAGE_NAME}
serviceaccount "pachyderm" created
persistentvolume "etcd-volume" created
persistentvolumeclaim "etcd-storage" created
replicationcontroller "etcd" created
service "etcd" created
service "pachd" created
replicationcontroller "pachd" created
secret "google-secret" created
➔ 
 
(after undeploy and redeploy)
➔ 
➔ kubectl get all
NAME             READY     STATUS    RESTARTS   AGE
po/etcd-mk7tm    1/1       Running   0          31s
po/pachd-fq8zf   1/1       Running   2          30s

NAME       DESIRED   CURRENT   READY     AGE
rc/etcd    1         1         1         31s
rc/pachd   1         1         1         30s

NAME             CLUSTER-IP     EXTERNAL-IP   PORT(S)                       AGE
svc/etcd         10.3.240.152   <nodes>       2379:32464/TCP                31s
svc/kubernetes   10.3.240.1     <none>        443/TCP                       21m
svc/pachd        10.3.255.164   <nodes>       650:30650/TCP,651:30651/TCP   30s
➔
```

Then I tried to put a text file, get the file, and list files.  Indeed I do reproduce the above error:

```sh
➔ cat blah.txt
blah
➔ pachctl create-repo testbug
➔ pachctl list-repo
NAME                CREATED             SIZE                
testbug             3 seconds ago       0 B                 
➔ pachctl put-file testbug master /blah.txt -c -f blah.txt
➔ pachctl get-file testbug master /blah.txt
storage: object doesn't exist
➔ pachctl list-file testbug master /
storage: object doesn't exist
➔ pachctl list-repo
NAME                CREATED              SIZE                
testbug             About a minute ago   5 B                 
➔ 
```

**Comment by dwhitena** [comment]
Another user hitting this issue:

> This problem popped up when I upgraded to 1.4, it did this before using minio too but I haven't delved into it as much (just trying it without minio again to get the kube event log and see what the result is)
> `pachctl list-repo` shows what I expect, input repo with a size > 0 and the pipeline output repos with sizes of 0
> `pachctl list-commit` shows the commit I made
> `pachctl list-file` with the repo/commit shows `object not found` for a google bucket or a similar error for minio

@derekchiang were you able to find a path forward on this yesterday?

**Comment by Southclaws** [comment]
Some extra info in case it helps, this is happening on our GKE cluster with Kubernetes 1.5.4, Pachyderm 1.4 tried with both a Google storage bucket and a custom backend (temporary Minio server running inside Kubernetes). The cluster node-pool has "storage-rw" scope.

A 1-file commit successfully went through (StartCommit, PutFileURL, FinishCommit) with no errors thrown.

Pipeline pods show up and describing shows:

```
Name:           pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c
Namespace:      spotlight
Node:           gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw/10.132.0.5
Start Time:     Tue, 04 Apr 2017 14:20:49 +0200
Labels:         app=pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1
                suite=pachyderm
Status:         Running
IP:             10.4.0.28
Controllers:    ReplicationController/pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1
Init Containers:
  init:
    Container ID:       docker://92d2a9d079acf844153817911443cb6bee4112ed2f8e0c296aa6d1b179ce5e02
    Image:              pachyderm/worker:1.4.0
    Image ID:           docker://sha256:c36ccffc2150b478467e4fc1ac8f2957563abd485edb138fc441c758882d1eca
    Port:               
    Command:
      /pach/worker.sh
    State:              Terminated
      Reason:           Completed
      Exit Code:        0
      Started:          Tue, 04 Apr 2017 14:20:50 +0200
      Finished:         Tue, 04 Apr 2017 14:20:50 +0200
    Ready:              True
    Restart Count:      0
    Volume Mounts:
      /pach-bin from pach-bin (rw)
      /var/pachyderm_worker from pachyderm-worker (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tnl33 (ro)
    Environment Variables:
      INPUT_DIRECTORY:          /pfs/jc2736f0e9b964200bf70f7cb047813adinput
      OUTPUT_DIRECTORY:         /pfs/out
      PPS_WORKER_IP:             (v1:status.podIP)
      PPS_POD_NAME:             pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c (v1:metadata.name)
      PPS_ETCD_PREFIX:          pachyderm_pps
      PPS_PIPELINE_NAME:        jc2736f0e9b964200bf70f7cb047813adhelloworldgo
Containers:
  user:
    Container ID:       docker://24792556c0ea352d78edbf5e8e91e24e0daa1b29e5ce9c42e9059930a65e550f
    Image:              spotlightdata/worker_helloworld_go:latest
    Image ID:           docker://sha256:ad88b46c20a2e5273bd22171ba75b3bb416c164aeee65227a8be072608249105
    Port:               
    Command:
      /pach-bin/guest.sh
    State:              Waiting
      Reason:           CrashLoopBackOff
    Last State:         Terminated
      Reason:           ContainerCannotRun
      Exit Code:        127
      Started:          Tue, 04 Apr 2017 14:26:38 +0200
      Finished:         Tue, 04 Apr 2017 14:26:38 +0200
    Ready:              False
    Restart Count:      6
    Volume Mounts:
      /pach-bin from pach-bin (rw)
      /var/pachyderm_worker from pachyderm-worker (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tnl33 (ro)
    Environment Variables:
      INPUT_DIRECTORY:          /pfs/jc2736f0e9b964200bf70f7cb047813adinput
      OUTPUT_DIRECTORY:         /pfs/out
      PPS_WORKER_IP:             (v1:status.podIP)
      PPS_POD_NAME:             pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c (v1:metadata.name)
      PPS_ETCD_PREFIX:          pachyderm_pps
      PPS_PIPELINE_NAME:        jc2736f0e9b964200bf70f7cb047813adhelloworldgo
Conditions:
  Type          Status
  Initialized   True 
  Ready         False 
  PodScheduled  True 
Volumes:
  pach-bin:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
  pachyderm-worker:
    Type:       HostPath (bare host directory volume)
    Path:       /var/pachyderm_worker
  default-token-tnl33:
    Type:       Secret (a volume populated by a Secret)
    SecretName: default-token-tnl33
QoS Class:      BestEffort
Tolerations:    <none>
Events:
  FirstSeen     LastSeen        Count   From                                                            SubObjectPath                   Type            Reason          Message
  ---------     --------        -----   ----                                                            -------------                   --------        ------          -------
  10m           10m             1       {default-scheduler }                                                                            Normal          Scheduled       Successfully assigned pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c to gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.initContainers{init}       Normal          Pulled          Container image "pachyderm/worker:1.4.0" already present on machine
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.initContainers{init}       Normal          Created         Created container with docker id 92d2a9d079ac; Security:[seccomp=unconfined]
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.initContainers{init}       Normal          Started         Started container with docker id 92d2a9d079ac
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}           Normal          Pulling         pulling image "spotlightdata/worker_helloworld_go:latest"
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}           Normal          Pulled          Successfully pulled image "spotlightdata/worker_helloworld_go:latest"
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}           Normal          Created         Created container with docker id 53caae63c8aa; Security:[seccomp=unconfined]
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}           Warning         Failed          Failed to start container with docker id 53caae63c8aa with error: Error response from daemon: Container command '/pach-bin/guest.sh' not found or does not exist.
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}           Normal          Created         Created container with docker id b84052c18d7d; Security:[seccomp=unconfined]
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}           Warning         Failed          Failed to start container with docker id b84052c18d7d with error: Error response from daemon: Container command '/pach-bin/guest.sh' not found or does not exist.
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}           Normal          Created         Created container with docker id 515e292eced5; Security:[seccomp=unconfined]
  10m           10m             1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}           Warning         Failed          Failed to start container with docker id 515e292eced5 with error: Error response from daemon: Container command '/pach-bin/guest.sh' not found or does not exist.
  10m           10m             2       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}                                       Warning         FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "user" with CrashLoopBackOff: "Back-off 20s restarting failed container=user pod=pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c_spotlight(23910131-1931-11e7-b11f-42010a84000c)"

  9m    9m      1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Warning Failed          Failed to start container with docker id c07ea39c038f with error: Error response from daemon: Container command '/pach-bin/guest.sh' not found or does not exist.
  9m    9m      1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Normal  Created         Created container with docker id c07ea39c038f; Security:[seccomp=unconfined]
  9m    9m      2       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}                               Warning FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "user" with CrashLoopBackOff: "Back-off 40s restarting failed container=user pod=pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c_spotlight(23910131-1931-11e7-b11f-42010a84000c)"

  9m    9m      1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Warning Failed          Failed to start container with docker id 518f0de9c117 with error: Error response from daemon: Container command '/pach-bin/guest.sh' not found or does not exist.
  9m    9m      1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Normal  Created         Created container with docker id 518f0de9c117; Security:[seccomp=unconfined]
  9m    7m      7       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}                               Warning FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "user" with CrashLoopBackOff: "Back-off 1m20s restarting failed container=user pod=pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c_spotlight(23910131-1931-11e7-b11f-42010a84000c)"

  7m    7m      1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Warning Failed          Failed to start container with docker id 09adc5112006 with error: Error response from daemon: Container command '/pach-bin/guest.sh' not found or does not exist.
  7m    7m      1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Normal  Created         Created container with docker id 09adc5112006; Security:[seccomp=unconfined]
  7m    4m      12      {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}                               Warning FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "user" with CrashLoopBackOff: "Back-off 2m40s restarting failed container=user pod=pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c_spotlight(23910131-1931-11e7-b11f-42010a84000c)"

  4m    4m      1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Warning Failed          Failed to start container with docker id 24792556c0ea with error: Error response from daemon: Container command '/pach-bin/guest.sh' not found or does not exist.
  10m   4m      6       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Normal  Pulled          Container image "spotlightdata/worker_helloworld_go:latest" already present on machine
  10m   4m      7       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}                               Warning FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "user" with RunContainerError: "runContainer: Error response from daemon: Container command '/pach-bin/guest.sh' not found or does not exist."

  4m    4m      1       {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Normal  Created         Created container with docker id 24792556c0ea; Security:[seccomp=unconfined]
  10m   4s      44      {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}       spec.containers{user}   Warning BackOff         Back-off restarting failed docker container
  4m    4s      21      {kubelet gke-spotlight-cluster-node-pool-scaling-8b3313b5-0cbw}                               Warning FailedSync      Error syncing pod, skipping: failed to "StartContainer" for "user" with CrashLoopBackOff: "Back-off 5m0s restarting failed container=user pod=pipeline-jc2736f0e9b964200bf70f7cb047813adhelloworldgo-v1-17n0c_spotlight(23910131-1931-11e7-b11f-42010a84000c)"
```

`pachctl list-repo` shows the repos and the input repo has bytes in it. `pachctl list-commit` with the input repo shows that something went in successfully.
```
REPO                                     ID                                 PARENT              STARTED             DURATION            SIZE
j9dd08b64955042acba8baf8e6fdf7377input   1b47631e4b584222b8a8533316c7ab66   <none>              42 seconds ago      2 seconds           59.38 KiB
```
but `pachctl list-file` just errors out with `storage: object doesn't exist`.

**Comment by dwhitena** [comment]
Fixed in 1.4.1.
